engine/storage/snapshot/src/org/apache/cloudstack/storage/snapshot/XenserverSnapshotStrategy.java

```diff
@@ -374,8 +375,24 @@ public SnapshotInfo takeSnapshot(SnapshotInfo snapshot) {
 
             snapshot = result.getSnashot();
             DataStore primaryStore = snapshot.getDataStore();
+            boolean backupFlag = Boolean.parseBoolean(configDao.getValue(Config.BackupSnapshotAfterTakingSnapshot.toString()));
```

Comment:
@nathanejohnson in #1600 I've added a flag to the create snapshot API called `locationType` which can take `primary` or `secondary` as the parameter. We could use that here as well. Although for this PR, a global would suffice